### ISSUE TITLE
Duplication glitch

### ISSUE THREAD

**Issue by PaisWillie** [Rocologo/BagOfGold#93]
When I drop the bag of gold on the ground, walk a few blocks away, and walk back to pick it back up, it will duplicate its value 90% of the time.

Server info:
- Spigot
- 1.15.2
- Plugin version 2.9.6

**Comment by PaisWillie** [comment]
Debug info when:

1. I gave myself $1
2. I dropped it on the ground
3. I walked a bit away from it
4. I picked it up
5. Repeated from step 2 twice

```
27.04 17:49:06 [Server] Server thread/INFO WisdomGamer issued server command: /eco give WisdomGamer 1
27.04 17:49:06 [Server] Server thread/INFO [BagOfGold] [Debug] Check if WisdomGamer has 0 Bells on the balance=0)
27.04 17:49:06 [Server] Server thread/INFO [BagOfGold] [Debug] Withdraw 0 from WisdomGamer's account, new balance is 0
27.04 17:49:06 [Server] Server thread/INFO [BagOfGold] [Debug] EconomyManager: withdrawPlayer adjusting Player Balance to Amount of BagOfGold in Inventory
27.04 17:49:06 [Server] Server thread/INFO [BagOfGold] [Debug] WisdomGamer has room for 3599964.0 BagOfGold in the inventory
27.04 17:49:06 [Server] Server thread/INFO [BagOfGold] [Debug] There was no difference
27.04 17:49:06 [Server] Server thread/INFO [BagOfGold] [Debug] WisdomGamer has room for 3599964.0 BagOfGold in the inventory
27.04 17:49:06 [Server] Server thread/INFO [BagOfGold] [Debug] Deposit 1 to WisdomGamer's account, new balance is 1
27.04 17:49:08 [Server] Server thread/INFO [BagOfGold] [Debug] WisdomGamer dropped 1 Bells. (# of rewards left=2)(2)
27.04 17:49:08 [Server] Server thread/INFO [BagOfGold] [Debug] WisdomGamer dropped 1.0 Bells using Q key
27.04 17:49:08 [Server] Server thread/INFO [BagOfGold] [Debug] Removing 1 from WisdomGamer's balance 1
27.04 17:49:14 [Server] Server thread/INFO [BagOfGold] [Debug] Added 1 to WisdomGamer's item in slot 0, new value is 2 (addBagOfGoldPlayer_EconomyManager)
27.04 17:49:23 [Server] Server thread/INFO [BagOfGold] [Debug] WisdomGamer dropped 2 Bells. (# of rewards left=2)(2)
27.04 17:49:23 [Server] Server thread/INFO [BagOfGold] [Debug] WisdomGamer dropped 2.0 Bells using Q key
27.04 17:49:23 [Server] Server thread/INFO [BagOfGold] [Debug] Removing 2 from WisdomGamer's balance 1
27.04 17:49:28 [Server] Server thread/INFO [BagOfGold] [Debug] Added 2 to WisdomGamer's item in slot 0, new value is 4 (addBagOfGoldPlayer_EconomyManager)
```

**Comment by PaisWillie** [comment]
My balances still registers as only dropping and picking up 1 gold, until I open my inventory, left click the gold to pick it up, and left click it again to drop it, then it updates my balance to whatever the value it says on the skull, which was 4, as I duplicated it twice.
```
27.04 17:51:47 [Server] MH TaskThread/INFO [BagOfGold] [Debug] DatabaseDataStore: savedata: PlayerBalance: {player: Name:WisdomGamer, WorldGrp:default, GameMode:SURVIVAL, Balance: 0.0(+0.0), BankBalance: 0.0(+0.0)}
27.04 17:51:47 [Server] MH TaskThread/INFO [BagOfGold] [Debug] DatabaseDataStore: savedata: PlayerBalance: {player: Name:WisdomGamer, WorldGrp:default, GameMode:SURVIVAL, Balance: 1.0(+0.0), BankBalance: 0.0(+0.0)}
27.04 17:51:47 [Server] MH TaskThread/INFO [BagOfGold] [Debug] DatabaseDataStore: savedata: PlayerBalance: {player: Name:WisdomGamer, WorldGrp:default, GameMode:SURVIVAL, Balance: -1.0(+0.0), BankBalance: 0.0(+0.0)}
27.04 17:51:47 [Server] MH TaskThread/INFO [BagOfGold] [Debug] DatabaseDataStore: savedata: PlayerBalance: {player: Name:WisdomGamer, WorldGrp:default, GameMode:SURVIVAL, Balance: 1.0(+0.0), BankBalance: 0.0(+0.0)}
```

(I renamed the gold to Bells)

**Comment by PaisWillie** [comment]
```
AcceptTheRules v2.3
ActionAnnouncer v1.15.0
AdvancedFood v3.6.4
AdvancedRegionMarket v3.0.2
AnimatedFrames v4.11.5-SNAPSHOT
AntiXrayHeuristics v1.2.0
ArmorStandTools v3.5.1
AsyncWorldEdit v3.6.19
BagOfGold v2.9.6
BannerBoard v1.10.3
BuycraftX v12.0.6
Citizens v2.0.26-SNAPSHOT (build 1854)
CitizensCMD v2.4
ClearLag v3.1.6
CommandButtons v4.2.1
CommandSpy v2.0
CoreProtect v2.18.2
DeathMessagesPrime v1.15.3
DeluxeChat v1.15.0
DeluxeCommands v1.10.0
DeluxeJoin v2.8.0
DeluxeMentions v3.0.6
DeluxeTags v1.8.1
DiscordSRV v1.18.4
EpicRename v3.7.1
Essentials v2.17.2.0
EssentialsAntiBuild v2.17.2.0
EssentialsProtect v2.17.2.0
EssentialsSpawn v2.17.2.0
FabledSkyBlock v2.2.8
FeatherBoard v4.25.0
FogusCore v1.9.9
GrantX v1.4.0
GSit v[2.12.0.0]
HeadDatabase v4.12.1
HeapDump v1.4.0
HolographicChat v0.1
HolographicDisplays v2.4.1
HolographicPlaceholders v2.0.1 Beta
InsaneShops v1.1.9
InteractionVisualizer v1.10.7
InteractiveChat v2.0.5
LeaderHeadsRevamped v4.0.5
LightAPI v3.4.2
LimitPillagers v2.12.0
LiteBans v2.4.7
LuckPerms v5.0.94
LuckPermsGUI v3.5
MapManager v1.7.8-SNAPSHOT
Matrix v4.7.8
mcMMO v2.1.124
MessageAnnouncer v1.11.0
MobFarmManager v1.5.5.1
MovingDevApi v1.1.3
Multiverse-Core v4.1.0-b775
Multiverse-Inventories v3.0.0-b459
Multiverse-NetherPortals v4.1.0-b763
Multiverse-Portals v4.1.0-b805
MVdWPlaceholderAPI v3.0.1
OnTime v4.1.4
OreoCore v14.7
PacketListenerApi v3.7.5-SNAPSHOT
PlaceholderAPI v2.10.5
PlayerReport v3.2.12
PlugMan v2.1.7
PremiumVanish v2.5.11
ProtocolLib v4.5.0
RewardPRO v4.2.3
SelectionVisualizer v3.0.0.7
ServerTutorialPlus v1.21.3
SilkSpawners v6.0.0-SNAPSHOT-b${BUILD_NUMBER}
Skript v2.4.1
TAB v2.7.4
TaskScheduler v1.0
TokenManager v3.2.5
TradeSystem v1.1.8
TrMenu v1.21
UltimateStacker v1.11.7
Vault v1.7.2-b107
ViaBackwards v3.0.0-SNAPSHOT
ViaVersion v3.0.0-SNAPSHOT
VillagerOptimiser v1.1.2
Votifier v2.6.0-SNAPSHOT
Voting v2.5.0
VotingPlugin v5.21.1
WorldBorder v1.9.10 (beta)
WorldEdit v7.1.0;8e55131
WorldGuard v7.0.2-SNAPSHOT;bf2593e
WorldGuardExtraFlags v4.1.4
XProtect v2.5.1

```

**Comment by Rocologo** [comment]
Its very strange. The thing you is very basic and I have tested it more time than you can imagine. 

Think about the problem I got an idea, which might help, although I still dont understand whats wrong. Could you please test this build:

https://fractal.lindegaard.one:8181/job/BagOfGold/394/

You have alot of plugins on your server and it could be a conflict with one of them. Im not able to test all these plugins. One thin though I want to test, is that you use Multiverse-Inventories v3.0.0-b459, and I am using PerWorldInventory. I dont think this is the problem, but I will test anyway because I want to be sure that BagOfGold is working with Multiverse-Inventories as well.

**Comment by Rocologo** [comment]
I noticed one thing more in the debug log

```
27.04 17:49:06 [Server] Server thread/INFO WisdomGamer issued server command: /eco give WisdomGamer 1
27.04 17:49:06 [Server] Server thread/INFO [BagOfGold] [Debug] Check if WisdomGamer has 0 Bells on the balance=0)

```


It looks like "WisdomGamer" is an Essentials Eco admin? This means that When you use "/eco give WisdomGamer 1" with a balance of 0, this should not be allowed. So one Bell should not be dropped because balance is 0. 

I will follow this track, because I think this is the problem!

**Comment by PaisWillie** [comment]
**Here's what I tried, but didn't fix the bug:**

1. /plugman disable Multiverse-Inventories 
2. Deleting Multiverse-Inventories JAR file and restarting the server
3. Tried duplicating the glitch while in survival mode instead of creative mode
4. Tried duplicating the glitch using an alt account that was non-OP'd

**Then I tried this:**

1. Installed the test build JAR file and restarted server (Without Multiverse-Inventories) and the bug is fixed!
2. Reinstalled Multiverse-Inventories and the bug is still gone!

Maybe Multiverse-Inventories was not the issue, but it's fixed now :)

You also fixed that bug where when you move when you drop the gold, it picks it back up immediately.
